xbmc/rendering/dx/DeviceResources.cpp

```diff
@@ -279,10 +281,15 @@ void DX::DeviceResources::CreateDeviceResources()
   // Create the Direct3D 11 API device object and a corresponding context.
   ComPtr<ID3D11Device> device;
   ComPtr<ID3D11DeviceContext> context;
+  ComPtr<IDXGIFactory1> dxgiFactory;
+  ComPtr<IDXGIAdapter1> pAdapter;
+
+  CreateDXGIFactory1(IID_IDXGIFactory1, &dxgiFactory);
+  dxgiFactory->EnumAdapters1(monindex, &pAdapter);
```

Comment:
no need to create a factory and adapter here. you can assign `m_adapter` in `SetMonitor` method instead of using static `monitor` (why monitor?)

Reply:
I don't have the knowledge to how kodi works in regards of settings, I don't know the methods and interfaces etc, the only variable that points to the right adapter in DeviceResources is monitor that is passed from another class.

Reply:
We don't need a quick solution. we need a proper solution. You have to learn how Kodi works to make solution proper. 
My suggestion is try to direct you to proper solution.